web/src/lib/components/shared-components/side-bar/recent-albums.svelte

```diff
@@ -4,13 +4,19 @@
   import { getAllAlbums, type AlbumResponseDto } from '@immich/sdk';
   import { handleError } from '$lib/utils/handle-error';
   import { t } from 'svelte-i18n';
+  import { userInteraction } from '$lib/stores/user.svelte';
 
   let albums: AlbumResponseDto[] = $state([]);
```

Comment:
You could completely drop this variable and replace it with the global `userInteraction.recentAlbums` state

Reply:
Do you mean dropping `let albums: AlbumResponseDto[] = $state([]);` in favor of using only `userInteraction.recentAlbums`? If so, you have to wrap the each section in a `{#if userInteraction.recentAlbums}` because `userInteraction.recentAlbums` can be undefined. What do you prefer?

Reply:
Can it be undefined? You'd still set it in `onMount` if it's undefined, and `onMount` finishes before it actually mounts, right?

Reply:
Or is my svelte knowledge lacking here? 😅

Reply:
Yes, technically it can't be indefinite, but Svelte isn't smart enough to figure that out. It only uses the type definition

![image](https://github.com/user-attachments/assets/c92dd2ab-cea4-4ab2-9ae8-9680ff58267a)


```ts
interface UserInteractions {
  recentAlbums?: AlbumResponseDto[];
  versions?: ServerVersionHistoryResponseDto[];
  aboutInfo?: ServerAboutResponseDto;
  serverInfo?: ServerStorageResponseDto;
}
```

Reply:
Ah gotcha. Then I'd prefer keeping that extra variable as well